src/index.ts

```diff
@@ -1,13 +1,25 @@
 import './set-public-path';
+import { ExtensionDefinition, attach } from '@openmrs/esm-extension-manager';
 import { backendDependencies } from './openmrs-backend-dependencies';
 
 const importTranslation = require.context('../translations', false, /.json$/, 'lazy');
 
+const extensions: Array<ExtensionDefinition> = [
+  {
+    appName: '@openmrs/esm-patient-registration-app',
+    name: 'registration-link',
+    load: () => import('./registration-link'),
+  },
+];
+
 function setupOpenMRS() {
   return {
     lifecycle: () => import('./openmrs-esm-patient-registration'),
     activate: 'patient-registration',
+    extensions: extensions,
   };
 }
 
+attach('home-page-buttons', 'registration-link');
```

Comment:
@brandones , I'm a little surprised by this. Does this mean that the extension needs to know where it goes? I was under the impression that extensions themselves would have no knowledge of where they would be placed. The matching of extension to extension slot would be handled by a configuration file belonging to the module defining the extension slots.

Reply:
@jonathandick sure

Reply:
Extensions don’t *need* to know where they get plugged in, but they can. This is how you get things to magically appear when you turn a module on.

The attachment can also be made in code on the extension slot side.

Any of these hardcoded attachments can be removed or reordered or configured in config. Config can also create new attachments that were not hardcoded.

So in summary—hard coding just allows us to make the application do reasonable things by default. The config gives full control.

Reply:
Just to restate what I said on the call for posterity—
1. The Implementer Tools UI Editor should make it obvious what's attached where and what comes from what
2. We can make an [esm-cli](https://github.com/openmrs/openmrs-esm-cli) documentation generator which lists `attach`ments to the README

Reply:
Sounds good to me.